anker/src/state.rs

```diff
@@ -836,4 +904,52 @@ mod test {
             assert_eq!(anker, anker_recovered);
         }
     }
+
+    #[test]
+    fn test_historical_price_array_minimum() {
+        let mut price_array = HistoricalStSolPriceArray::new();
+        // 100 UST for each StSol.
+        for slot in 0..POOL_PRICE_NUM_SAMPLES as u64 {
+            price_array.insert_and_rotate(slot, MicroUst(100_000_000));
+        }
+
+        // 1 StSol rewards and 1% slippage.
+        let minimum_ust = price_array
+            .calculate_minimum_price(StLamports(1_000_000_000), 9900)
+            .unwrap();
+        assert_eq!(minimum_ust, MicroUst(99_000_000));
+
+        // 1 StSol rewards and 2% slippage.
+        let minimum_ust = price_array
+            .calculate_minimum_price(StLamports(1_000_000_000), 9800)
+            .unwrap();
+        assert_eq!(minimum_ust, MicroUst(98_000_000));
+
+        // 80 StSol rewards and 5% slippage
+        let minimum_ust = price_array
+            .calculate_minimum_price(StLamports(80_000_000_000), 9500)
+            .unwrap();
+        assert_eq!(minimum_ust, MicroUst(7_600_000_000));
+
+        // 331 StSol rewards and 50% slippage
+        let minimum_ust = price_array
+            .calculate_minimum_price(StLamports(331_000_000_000), 5000)
+            .unwrap();
+        assert_eq!(minimum_ust, MicroUst(16_550_000_000));
+    }
```

Comment:
This test is good, but it tests the very special case where all samples are the same. We should have an independent test to confirm that finding the median works.

Also, we should have a test to confirm that `insert_and_rotate` kicks out the right value.